packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -313,6 +314,7 @@ function hydrateCustomElement(
     }
 
     const { sel, mode, ctor, owner } = vnode;
+    renderer.defineCustomElement(StringToLowerCase.call(elm.tagName));
```

Comment:
`engine-core` should not call `elm.tagName` directly – we have `renderer.getTagName` to abstract that.